File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06IO2/CommandTest/CommandTest/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommandTest
{
    class Sportsmen
    {
        private string name;
        public Sportsmen(string name) { this.name = name; } 
        public string run() { return "Running"; }
        public string jump() { return "Jumping"; }
        public string swim() { return "Swimming"; }
        public string execute(SportsmenCommand cmd) { return name+": "+cmd.execute(this); }
    }

    interface SportsmenCommand
    {
        string execute(Sportsmen s);
    }

    class SportsmenRun : SportsmenCommand
    {
        public string execute(Sportsmen s) { return s.run(); }
        public override string ToString() { return "Run"; }
    }
    class SportsmenJump : SportsmenCommand
    {
        public string execute(Sportsmen s) { return s.jump(); }
        public override string ToString() { return "Jump"; }
    }
    class SportsmenSwim : SportsmenCommand
    {
        public string execute(Sportsmen s) { return s.swim(); }
        public override string ToString() { return "Swim"; }
    }
    interface SportsmenComplexCommand : SportsmenCommand
    {
    }
    class SportsmenMultiplyCommand : SportsmenComplexCommand
    {
        private SportsmenCommand cmd;
        private int count;
        public SportsmenMultiplyCommand(int count, SportsmenCommand cmd)
        {
            this.count = count;
            this.cmd = cmd;
        }
        public string execute(Sportsmen s)
        {
            StringBuilder sb = new StringBuilder();
            return string.Format("{0} x ({1})", count, cmd.execute(s));
        }
        public override string ToString() { return count+"x"+cmd; }
    }
    class SportsmenSequenceCommand : SportsmenComplexCommand
    {
        private SportsmenCommand[] tb;
        private string name;
        public SportsmenSequenceCommand(string name,SportsmenCommand[] tb)
        {
            this.name = name;
            this.tb = tb;
        }
        public string execute(Sportsmen s)
        {
            StringBuilder sb = new StringBuilder();
            bool first=true;
            foreach (SportsmenCommand cmd in tb)
            {
                if (first) first = false;
                else sb.Append(", ");
                sb.Append(cmd.execute(s));
            }
            return sb.ToString();
        }
        public override string ToString() { return name; }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2020s/IZ08IO1/CommandTest/src/Sportsmen.java
public class Sportsmen 
{
    private final String name;
    public Sportsmen(String name) { this.name=name; }
    public String run() { return "Running"; }
    public String jump() { return "Jumping"; }
    public String swim() { return "Swimming"; }
    public String execute(Command cmd) { return name+": "+cmd.execute(this); }
    //@Override public String toString() { return name; }    
}

interface Command
{
    String execute(Sportsmen s);
}

class CommandRun implements Command
{
    public String execute(Sportsmen s) { return s.run(); }
    @Override public String toString() { return "Run"; }
}

class CommandJump implements Command
{
    public String execute(Sportsmen s) { return s.jump(); }
    @Override public String toString() { return "Jump"; }
}

class CommandSwim implements Command
{
    public String execute(Sportsmen s) { return s.swim(); }
    @Override public String toString() { return "Swim"; }
}

class CommandTrening implements Command
{
    private final String name;
    private final Command[] lst;
    public CommandTrening(String name, Command[] lst) 
    {
        this.name=name;
        this.lst=lst;
    }    
    public String execute(Sportsmen s)
    {
        final StringBuilder sb=new StringBuilder();
        boolean first=true;
        for(Command cmd:lst)
        {
            if(first) first=false; else sb.append(", ");
            sb.append(cmd.execute(s));
        }
        return sb.toString();
    }
    @Override public String toString() { return name; }
}
